### PR TITLE
Add structs and interfaces for SM cpp implementation

### PR DESCRIPTION
**Pull request by mlohvynenko**

### CODE REVIEW

File: include/aos/sm/launcher.hpp [hunk 1]
```diff
@@ -45,6 +45,24 @@ class LauncherItf {
...
+     *
+     * @param envVarsInfo environment variables info.
+     * @param statuses[out] environment variables statuses.
+     * @return Error
+     */
+    virtual Error OverrideEnvVars(const Array<EnvVarsInstanceInfo>& envVarsInfo, EnvVarsInstanceStatusArray& statuses)
+        = 0;
+
+    /**
+     * Set cloud connection status.
```
Comment (al1img) [inline include/aos/sm/launcher.hpp:68]: Sets

File: include/aos/sm/networkmanager.hpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+     * @param instanceID instance id.
+     * @param[out] inputTraffic instance's input traffic.
+     * @param[out] outputTraffic instance's output traffic.
+     * @return Error.
+     */
+    virtual Error GetInstanceTraffic(const String& instanceID, uint64_t& inputTraffic, uint64_t& outputTraffic) const
+        = 0;
+
+    /**
+     * Set the traffic period.
```
Comment (al1img) [inline include/aos/sm/networkmanager.hpp:91]: Sets

File: include/aos/common/types.hpp [hunk 3]
```diff
@@ -1066,6 +1091,132 @@ struct EnvVarsInstanceStatus {
 
 using EnvVarsInstanceStatusArray = StaticArray<EnvVarsInstanceStatus, cMaxNumInstances>;
 
+/**
```
Comment (al1img) [inline include/aos/common/types.hpp:1094]: Move to cloudprotocol namespace

File: include/aos/sm/logprovider.hpp [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+namespace logprovider {
+
+/** @addtogroup sm Service Manager
+ *  @{
+ */
+
+/**
+ * Logs observer interface.
+ */
+class LogsObserverItf {
```
Comment (al1img) [inline include/aos/sm/logprovider.hpp:24]: LogObserverItf

File: include/aos/sm/logprovider.hpp [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+namespace aos {
+namespace sm {
+namespace logprovider {
+
+/** @addtogroup sm Service Manager
+ *  @{
+ */
+
+/**
+ * Logs observer interface.
```
Comment (al1img) [inline include/aos/sm/logprovider.hpp:22]: Log observer interface.

File: include/aos/sm/launcher.hpp [hunk 6]
```diff
@@ -30,6 +30,14 @@ namespace launcher {
  *  @{
  */
 
+/**
+ * Launcher operation version.
+ * If new functionality doesn't allow existing services to work
+ * properly, this value should be increased. It will force to remove all
+ * services and their storages before first start.
+ */
+constexpr auto cOperationVersion = AOS_CONFIG_LAUNCHER_OPERATION_VERSION;
```
Comment (al1img) [inline include/aos/sm/launcher.hpp:39]: It is pure const and isn't supposed to be changed with config. Make it either public const of Launcher or private const and add GetOperationVersion method to launcher itf.

File: include/aos/common/alerts/alerts.hpp [hunk 7]
```diff
@@ -0,0 +1,460 @@
...
+
+/**
+ * Resource alert errors len.
+ */
+constexpr auto cAlertResourceErrorsLen = AOS_CONFIG_ALERT_RESOURCE_ERRORS_LEN;
+
+/**
+ * System alert tag.
+ */
+constexpr auto cSystemAlertTag = "systemAlert";
```
Comment (al1img) [inline include/aos/common/alerts/alerts.hpp:69]: Make aos enum for alert tags

File: include/aos/common/alerts/alerts.hpp [hunk 8]
```diff
@@ -0,0 +1,460 @@
...
+
+/**
+ * Service instance alert tag.
+ */
+constexpr auto cServiceInstanceAlertTag = "serviceInstanceAlert";
+
+/**
+ * Alert item.
+ */
+struct AlertItem {
```
Comment (al1img) [inline include/aos/common/alerts/alerts.hpp:109]: Move cloud related structs to cloudprotocol namespace.

File: include/aos/common/alerts/alerts.hpp [hunk 9]
```diff
@@ -0,0 +1,460 @@
...
+     * @param timestamp alert timestamp.
+     */
+    AlertItem(const String& tag, const Time& timestamp)
+        : mTimestamp(timestamp)
+        , mTag(tag)
+    {
+    }
+
+    Time                       mTimestamp;
+    StaticString<cAlertTagLen> mTag;
```
Comment (al1img) [inline include/aos/common/alerts/alerts.hpp:123]: Should be aos enum

File: include/aos/common/cloudprotocol/envvars.hpp [hunk 10]
```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright (C) 2024 Renesas Electronics Corporation.
+ * Copyright (C) 2024 EPAM Systems, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef AOS_ENVVARS_HPP_
```
Comment (al1img) [inline include/aos/common/cloudprotocol/envvars.hpp:8]: AOS_CLOUDPROTOCOL_ENVVARS_HPP_

**Review by mykola-kobets-epam — APPROVED** [review]

File: include/aos/common/config.hpp [hunk 11]
```diff
@@ -148,6 +148,41 @@
...
+ */
+#ifndef AOS_CONFIG_TYPES_MAX_NUM_DNS_SERVERS
+#define AOS_CONFIG_TYPES_MAX_NUM_DNS_SERVERS 4
+#endif
+
+/**
+ * Max number of firewall rules.
+ */
+#ifndef AOS_CONFIG_TYPES_MAX_NUM_FIREWALL_RULES
+#define AOS_CONFIG_TYPES_MAX_NUM_FIREWALL_RULES 4
```
Comment (MykolaSolyanko) [inline include/aos/common/config.hpp:176]: i think in my opinion it should more than 4, lets use 10 and then will look

File: include/aos/common/cloudprotocol/alerts.hpp [hunk 12]
```diff
@@ -0,0 +1,479 @@
...
+     */
+    SystemQuotaAlert(const Time& timestamp = Time::Now())
+        : AlertItem(AlertTagEnum::eSystemQuotaAlert, timestamp)
+    {
+    }
+
+    StaticString<cNodeIDLen>         mNodeID;
+    StaticString<cAlertParameterLen> mParameter;
+    uint64_t                         mValue;
+    StaticString<cAlertStatusLen>    mStatus;
```
[the remark this answers is not in the document]
Comment (mlohvynenko) [reply include/aos/common/cloudprotocol/alerts.hpp:273]: Anyway gRPC needs it.
I'd leave the decision to parse/not parse to CM :)

File: include/aos/common/cloudprotocol/alerts.hpp [hunk 13]
```diff
@@ -0,0 +1,536 @@
...
+
+#ifndef AOS_CLOUDPROTOCOL_ALERTS_HPP_
+#define AOS_CLOUDPROTOCOL_ALERTS_HPP_
+
+#include "aos/common/tools/variant.hpp"
+#include "aos/common/types.hpp"
+
+namespace aos {
+namespace cloudprotocol {
+namespace alerts {
```
Comment (al1img) [inline include/aos/common/cloudprotocol/alerts.hpp:16]: We don't need additional to cloudprotocol namespace. Remove alerts namespace. All names already include alert prefix. For example:

```c++
aos::cloudprotocol::alerts::AlertItem
aos::cloudprotocol::AlertItem
```

And we agreed before to provide namespace per module: cloudprotocol this is module in this case.

File: include/aos/common/cloudprotocol/envvars.hpp [hunk 14]
```diff
@@ -0,0 +1,141 @@
...
+
+#ifndef AOS_CLOUDPROTOCOL_ENVVARS_HPP_
+#define AOS_CLOUDPROTOCOL_ENVVARS_HPP_
+
+#include "aos/common/cloudprotocol/cloudprotocol.hpp"
+#include "aos/common/types.hpp"
+
+namespace aos {
+namespace cloudprotocol {
+namespace envvars {
```
Comment (al1img) [inline include/aos/common/cloudprotocol/envvars.hpp:16]: remove envars namespace

File: include/aos/common/cloudprotocol/log.hpp [hunk 15]
```diff
@@ -0,0 +1,244 @@
...
+
+#include <cstdint>
+
+#include "aos/common/cloudprotocol/cloudprotocol.hpp"
+#include "aos/common/tools/optional.hpp"
+#include "aos/common/types.hpp"
+
+namespace aos {
+namespace cloudprotocol {
+namespace log {
```
Comment (al1img) [inline include/aos/common/cloudprotocol/log.hpp:19]: remove log namespace

**Review by al1img — APPROVED** [review]

**Pull request merged**